docs: how to submit annotations via API

TomasTomecek opened the pull request:
Fixes https://github.com/fedora-copr/logdetective/issues/219
Fixes #80

TomasTomecek commented:
@kwk please let me know if this works for you

If you do successful submissions, please let us know the IDs so we can check if they are valid, just to be sure :)

kwk commented on contribute-api.py:
```diff
@@ -0,0 +1,32 @@
+import requests
+
+build_id = "8896521"
+chroot = "rhel-9-x86_64"
+url = f"https://logdetective.com/frontend/contribute/copr/{build_id}/{chroot}"
+# url = f"http://localhost:5020/frontend/contribute/copr/{build_id}/{chroot}"
+data = {
+    "username": "FAS:me",
+    "fail_reason": "Failed because...",
+    "how_to_fix": "Like this...",
+    "spec_file": {
+        "name": "llvm.spec",
+        "content": "Yes, actual content of the spec file"
+    },
+    # if you are annotating a container build
+    # "container_file": None,
+    "logs": [
+        {
+            "name": "build.log",
+            "content": "content of the build log",
+            "snippets": [
+                {
+                    "start_index": 1,
+                    "end_index": 2,
+                    "user_comment": "this snippet is relevant because...",
+                    "text": "content of the snippet"
+                }
+            ]
+        }
+    ]
+}
+requests.post(url, json=data)
```
Why is this file added here? It looks likes the code from the documentation.

TomasTomecek replied:
That's what I was playing with, I committed it without thinking much about it.

kwk commented on README.md:
```diff
@@ -6,6 +6,46 @@ Production instance: http://logdetective.com
+    # "container_file": None,
+    "logs": [
+        {
+            "name": "build.log",
+            "content": "content of the build log",
```
Does it have to be a build.log or can it be a compressed version as well? If so, does `content` accept base64 or can one refer to the log and log-detective fetches it?

TomasTomecek replied:
Current codebase only accepts raw text right now. No base64, no link.

It can be any log file.

kwk replied:
@TomasTomecek let's take [this build](https://copr.fedorainfracloud.org/coprs/g/fedora-llvm-team/llvm-snapshots-big-merge-20250508/build/9009985/). The [chroot directory](https://download.copr.fedorainfracloud.org/results/@fedora-llvm-team/llvm-snapshots-big-merge-20250508/centos-stream-10-aarch64/09009985-llvm/) only contains a [builder-live.log.gz](https://download.copr.fedorainfracloud.org/results/@fedora-llvm-team/llvm-snapshots-big-merge-20250508/centos-stream-10-aarch64/09009985-llvm/builder-live.log.gz) file. When we do our own analysis that is tailored for LLVM and provide log-detective with results of this analysis we currently extract the `.gz` file and read it. Are we supposed to send over the file as `build.log` then? And are we supposed to send *all* the text or just the snippet?

kwk commented on README.md:
```diff
@@ -6,6 +6,46 @@ Production instance: http://logdetective.com
+            "content": "content of the build log",
+            "snippets": [
+                {
+                    "start_index": 1,
+                    "end_index": 2,
```
I assume these are line indexes. An index typically starts at 0 but if it is a line number I assume it starts at 1. Please clarify.

TomasTomecek replied:
I recall they are character indexes.
@FrostyX @nikromen can you help me answer here please? I'll update the docs then.

FrostyX replied:
Yes, exactly. This:

```
data["logs"][0]["content"][start_index:end_index+1]
```

should be equal to the `text` field.

kwk reviewed (commented):
@TomasTomecek thank you for creating this documentation I think it can be very helpful. But I do have some questions.

TomasTomecek commented:
I dropped the extra file and only left the README part. Added comments for the fields. @kwk please let me know if it's clear


@FrostyX thank you for giving me precise code! :)

nikromen commented:
It may be worth mentioning we have automatically generated API docs page from code documentation https://log-detective.com/docs 

however it was not intended to be official documentation so it's a bit mess :D (e.g. fields are not documented or some useless routes are not hidden)

TomasTomecek commented on README.md:
```diff
@@ -6,6 +6,51 @@ Production instance: http://logdetective.com
+
+build_id = "8896521"
+chroot = "rhel-9-x86_64"
+url = f"https://logdetective.com/frontend/contribute/copr/{build_id}/{chroot}"
```
now that @nikromen pointed me to the official API docs... (thank you!)

yes there is: https://log-detective.com/docs#/default/contribute_upload_file_frontend_contribute_upload_post
